feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -129,4 +133,23 @@ export class ConfiguratorAttributeBaseComponent {
   createFocusId(attributeId: string, valueCode: string): string {
     return `${attributeId}--${valueCode}--focus`;
   }
+
+  /**
+   * Get code from attribute.
+   * The code is not a mandatory attribute (since not available for VC flavour),
+   * still it is mandatory in the context of CPQ. Calling this method therefore only
+   * makes sense when CPQ is active. In case the method is called in the wrong context, an exception will
+   * be thrown
+   *
+   * @param Attribute
```

Comment:
According to JS doc 
@param {Configurator.Attribute} attribute - Attribute also see [here](https://jsdoc.app/tags-param.html)